### PR TITLE
feat(rspack): add ignoreWarnings option

### LINKED ISSUE #2979: `ignoreWarnings` for filter warnings
### What problem does this feature solve?

Tells Rspack to ignore specific warnings.

Support regex type and function type can be easily implemented with some work on js side, supporting object type can be separated to another issue/PR, which will need more refactor work on rust side.

### What does the proposed API of configuration look like?

https://webpack.js.org/configuration/other-options/#ignorewarnings

### PR DESCRIPTION
**Pull request by nagy-nabil**
Close #2979

- [x] add `ignoreWarnings` option in rspack options schema 
- [x] add  `compilation.hooks.processWarnings`
- [x] create ignoreWarnings plugin
- [x] activate the plugin when the user enter `ignoreWarnings` option 

### Known Limitations 

- eslint warnings get passed to the plugin as one error so with ignoreWarnings you got all eslint warnings or not

### CODE REVIEW

**Review by ahabhgk — APPROVED** [review]
LGTM, Thanks!

**Review by ahabhgk — CHANGES_REQUESTED** [review]

File: .changeset/olive-lobsters-tie.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@rspack/core": minor
```
[the remark this answers is not in the document]
Comment (nagy-nabil) [reply .changeset/olive-lobsters-tie.md]: updated it, thank you

**Comment by nagy-nabil** [comment]
@ahabhgk i updated the inline snapshot in defaults.unittest, am I missing anything else?

**Review by ahabhgk — APPROVED** [review]
LGTM, thanks!

**Pull request merged**